### PR TITLE
feat: add editor modal

### PR DESCRIPTION
**Pull request by Vi-cci**
This PR wires up the application to display modal windows inside the app, in order to select an editor, but allowing easy customization of the data and options inside said modal.
    

---

## Summary by cubic
Added an editor selection modal to the app, allowing users to choose their preferred editor before linking rules. The selected editor is now used to determine the target directory for rule installation.

- **New Features**
  - Modal window for editor selection with customizable options.
  - Editor choice is saved and shown in the UI, and used when linking rules.

---

**Add Modal-Driven Editor Selection and Editor-Aware Rule Linking to TUI/CLI**

This major PR introduces an extensible modal overlay system for the interactive TUI, enabling users to select their preferred code editor (e.g., Cursor, Windsurf) before linking rules. The selection is persisted in the UI state, displayed in info panels, and used to direct all rule symlinks into editor-specific directories, replacing the prior .cursor/rules convention. The PR covers the full integration of the modal system (using bubbletea-overlay), editor-aware changes throughout rule linking and unlinking logic, and ensures CLI, TUI, and test code paths support the new, editor-centric architecture while remaining backward compatible. Several best-practice updates (test coverage, type safety, separation of components) are included, along with dependency upgrades. Reviewers should be aware of modal system extensibility, state propagation, and the effects on user workflows and compatibility with preexisting setups.

**Key Changes:**
• Introduces a reusable modal/overlay system built on bubbletea-overlay (internal/ui/components/modal.go, internal/ui/manager.go).
• Adds an Editor Selection modal, allowing users to select among supported editors; selection is stored in ``UI`` model state.
• Changes rule-linking logic to direct symlinks to editor-specific directories (formerly only .cursor/rules), updating internal/linker/linker.go signature and logic.
• Updates ``CLI`` entrypoints and all ``TUI`` paths to work with the new modal manager and editor-specific logic.
• Propagates editor choice to info panels and linking status displays throughout the ``UI``.
• Updates test suites for the linker and model to verify correct handling of editor-centric paths and edge cases.
• New dependencies added (bubbletea-overlay, plus updates in go.mod/go.sum).

**Affected Areas:**
• ``UI`` modal/overlay management (internal/ui/components/, internal/ui/manager.go, internal/ui/model.go)
• Rule-linking core (internal/linker/linker.go) and helper logic (`UnlinkRule`, `IsRuleLinked`)
• ``CLI``/``TUI`` entrypoints (cmd/rule-tool/main.go)
• Unit and integration tests for linker, ``UI`` model
• Project dependencies (go.mod, go.sum)

**Potential Impact:**

**Functionality**: Users now select an editor before operating on rules in the TUI; all symlinks now respect editor folders (.cursor/rules, .windsurf/rules, etc.), affecting the location and management of linked rules for different editors. Existing links in .cursor/rules remain but are not affected by new selections.

**Performance**: Minimal impact due to lightweight overlays and localized state changes.

**Security**: Editor folder is passed as a parameter to linker logic; no direct validation, but usage is currently restricted to a fixed UI-driven set of values, mitigating traversal or injection risk.

**Scalability**: Modal and overlay architecture is extensible for additional editors or custom modal experiences without significant refactor.

**Review Focus:**
• Correct ``UI`` modal behavior: focus, dismissal (especially Escape/cancel handling), side-effect propagation from modal to background.
• Consistency and safety of editor folder usage in all path-building/linking operations-ensure no regressions or traversal.
• Backward compatibility: ``CLI`` mode, preexisting .cursor/rules content.
• Robustness of state handling for editor selection and display in ``UI``.
• Test completeness for all new and refactored behaviors.
• Code extensibility: modularity of modal system, separation of logic between manager, background, overlay, and components.

<details>
<summary><strong>Testing Needed</strong></summary>

• Verify modal can be opened, selections and cancellations handled via all expected keybindings (`e`, escape, enter, arrows).
• Test linking and unlinking rules for each available editor; inspect filesystem to confirm correct directory target.
• Switch editors during a ``TUI`` session and confirm info panel, status bar, and link/unlink logic consistently reflect new selection.
• Attempt edge cases: empty, unexpected, or malformed editor selection; invalid/unusual input handling in modals.
• Execute ``CLI`` with --link and --unlink commands to verify backward compatibility.
• Run all associated linker and model unit tests.

</details>

<details>
<summary><strong>Code Quality Assessment</strong></summary>

**internal/ui/components/editor_modal.go**: Minimal, direct instantiation, no immediate issues.

**internal/ui/manager.go**: Good orchestration; fgCmd side-effect could be lost due to conditional return (consider revisiting if modal commands grow more complex); current Init is a no-op but could discard future async work.

**internal/ui/model.go**: State handling sound, getEditorPathName logic could result in invalid dir names with future input (edge-case validation needed).

**internal/linker/linker.go**: Core changes for editor-aware operations are mostly correct; helper functions now handle editorFolder but need validation and broader coverage for future editor types.

**internal/linker/linker_test.go**: Tests updated; stronger coverage for unknown editors would be prudent.

**internal/ui/model_test.go**: Covers new editor path function.

**cmd/rule-tool/main.go**: Correctly delegates editor folder to linker, uses new Manager in TUI; minor code duplication but logic clear.

**internal/ui/components/modal.go**: Reusable, cleanly separates modal state and styling; small risk of panic if passed empty buttons, recommend additional validation.

</details>

<details>
<summary><strong>Best Practices</strong></summary>

**UI**:
• Component-based separation of modal logic and state
• Explicit typing of messages and commands
• Centralized state management for overlays

**Testing**:
• Comprehensive test updates for new arguments and paths

**Extensibility**:
• Modal and overlay designed to support future editors and modal types

**Type Safety**:
• Explicitly-typed Bubble Tea models/messages

</details>

<details>
<summary><strong>Possible Issues</strong></summary>

• If an unknown or malformed editor is somehow selected via UI or code injection, invalid directories (e.g., '.cursor (default)') may be created or linked to.
• Unvalidated editorFolder input across the linker boundary could be hazardous if wired differently in the future (currently UI-restricted).
• Potential for orphaned rules or inconsistent UI state if links exist in multiple editor folders without clear migration paths.
• Modal dismissal with Escape or empty selection needs to be verified; logic could be stricter.
• Some modal commands (side effects) might not propagate due to conditional dispatch (see manager.go).
• Possible crash if modal buttons are ever nil/empty (guard in code but deserves vigilance).

</details>

---
*This summary was automatically generated by @propel-code-bot*

## Summary by CodeRabbit

- **New Features**
  - Added support for selecting your preferred editor (e.g., Cursor or Windsurf) via a new modal dialog in the UI.
  - The selected editor is now displayed in the info panel and used when linking rules.
- **Improvements**
  - Enhanced UI with a modal overlay for smoother editor selection and interaction.
  - Info and help content in the UI are now more organized and informative.
- **Chores**
  - Updated dependencies to support new UI features.

### CODE REVIEW

File: internal/ui/manager.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+			0,
+			0,
+		),
+		showModal: false,
+	}
+}
+
+// Init initializes the Manager
+func (m *Manager) Init() tea.Cmd {
+	return nil
```
[the remark this answers is not in the document]
Comment (Vi-cci) [reply internal/ui/manager.go:35]: @cubic-dev-ai Right now we're not returning (or planning to return) any async commands from the background model, so this is correct right now.

File: internal/ui/model.go [hunk 2]
```diff
@@ -87,6 +97,7 @@ func New(cfg *config.Config, rulesManager *rules.Manager, linker *linker.Linker)
 		successMessage: "",
 		showingSuccess: false,
 		successTimer:   0,
+		editor:         "Cursor (default)", // Default editor value
```
[the remark this answers is not in the document]
Comment (Vi-cci) [reply internal/ui/model.go:100]: @cubic-dev-ai This is an extremely good suggestion that highlights an edge case that I missed. Thanks!

File: internal/linker/linker.go [hunk 3]
```diff
@@ -83,8 +83,8 @@ func (l *Linker) LinkRule(rule *models.Rule) error {
 		targetFileName = filepath.Base(rule.Path)
 	}
 
-	// Set the target path in the flat .cursor/rules directory
-	targetPath := filepath.Join(l.TargetDir, ".cursor", "rules", targetFileName)
+	// Set the target path in the specified editor's rules directory
+	targetPath := filepath.Join(l.TargetDir, editorFolder, "rules", targetFileName)
```
[the remark this answers is not in the document]
Comment (Vi-cci) [reply internal/linker/linker.go:87]: @cubic-dev-ai `editorFolder` is used exclusively for specifying the specific name of the folder for the editor that the user selected. Although nothing defends against an unvalidated and unsafe usage of this method, the way it's wired in right now prevents an scenario like you mention from ever happening. If the usage of this method changes drastically in the future, it might be appropriate to consider a validation step previous to just using the path as-is.
